app/code/Magento/InventorySourceSelectionApi/Exception/UndefinedInventoryRequestBuilderException.php

```diff
@@ -0,0 +1,15 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySourceSelectionApi\Exception;
+
+use Magento\Framework\Exception\LocalizedException;
+
+class UndefinedInventoryRequestBuilderException extends LocalizedException
```

Comment:
We don't recommend to introduce custom exception types if there is no really good reason of doing this.

In this case I would better to use `\Magento\Framework\Exception\InputException` as we do in similar cases